pkg/controller/node/node_controller.go

```diff
@@ -488,6 +495,10 @@ func (ctrl *Controller) updateNode(old, cur interface{}) {
 			ctrl.logPoolNode(pool, curNode, "changed labels")
 			changed = true
 		}
+		if !reflect.DeepEqual(oldNode.Spec.Taints, curNode.Spec.Taints) {
```

Comment:
prob need a nil check here?

Reply:
Why would one be non-nil empty and other be nil but I am if it is confusing, I can have `len()` checks as well.

Reply:
I doubt spec will ever be nil at this point in time, as for `Taints` `DeepEqual` handles nil case, for you.

Reply:
That is correct. Technically, the code above is right. I believe @kikisdeliveryservice's concern is more on the lines of readability.